toolkit/tools/specreader/specreader.go

```diff
@@ -26,78 +30,192 @@ const (
 	defaultWorkerCount = "10"
 )
 
+// parseResult holds the worker results from parsing a SPEC file.
+type parseResult struct {
+	packages []*pkgjson.Package
+	err      error
+}
+
 var (
-	app      = kingpin.New("specreader", "A tool to parse spec dependencies into JSON")
-	dir      = exe.InputDirFlag(app, "Directory to scan for SPECS")
-	output   = exe.OutputFlag(app, "Output file to export the JSON")
-	workers  = app.Flag("workers", "Number of concurrent goroutines to parse with").Default(defaultWorkerCount).Int()
-	srpmDir  = app.Flag("srpm-dir", "Directory containing SRPMs.").Required().ExistingDir()
-	macroDir = app.Flag("macro-dir", "Directory containing rpm macros.").Default("").String()
-	distTag  = app.Flag("dist-tag", "The distribution tag the SPEC will be built with.").Required().String()
-	logFile  = exe.LogFileFlag(app)
-	logLevel = exe.LogLevelFlag(app)
+	app       = kingpin.New("specreader", "A tool to parse spec dependencies into JSON")
+	specsDir  = exe.InputDirFlag(app, "Directory to scan for SPECS")
+	output    = exe.OutputFlag(app, "Output file to export the JSON")
+	workers   = app.Flag("workers", "Number of concurrent goroutines to parse with").Default(defaultWorkerCount).Int()
+	buildDir  = app.Flag("build-dir", "Directory to store temporary files while parsing.").String()
+	srpmsDir  = app.Flag("srpm-dir", "Directory containing SRPMs.").Required().ExistingDir()
+	rpmsDir   = app.Flag("rpm-dir", "Directory containing built RPMs.").Required().ExistingDir()
+	distTag   = app.Flag("dist-tag", "The distribution tag the SPEC will be built with.").Required().String()
+	workerTar = app.Flag("worker-tar", "Full path to worker_chroot.tar.gz").ExistingFile()
```

Comment:
Mention that if this argument is empty it'll be parsed with the host environment.